sdk/storage/Azure.Storage.DataMovement/tests/TestEventsRaised.cs

```diff
@@ -0,0 +1,266 @@
+            Assert.AreEqual(StorageTransferStatus.InProgress, StatusEvents.First().StorageTransferStatus);
+            Assert.AreEqual(StorageTransferStatus.Completed, StatusEvents.ElementAt(1).StorageTransferStatus);
+        }
+
+        /// <summary>
+        /// This asserts that the expected events occurred during a single transfer that is expected
+        /// to have a <see cref="StorageTransferStatus.CompletedWithSkippedTransfers"/> at the end without any
+        /// or failures.
+        /// </summary>
+        public void AssertSingleSkippedCheck()
+        {
+            AssertUnexpectedFailureCheck();
+            Assert.IsEmpty(SingleCompletedEvents);
+            Assert.AreEqual(2, StatusEvents.Count);
+            Assert.AreEqual(1, SkippedEvents.Count);
+            Assert.AreEqual(StorageTransferStatus.InProgress, StatusEvents.First().StorageTransferStatus);
+            Assert.AreEqual(StorageTransferStatus.CompletedWithSkippedTransfers, StatusEvents.ElementAt(1).StorageTransferStatus);
+            Assert.NotNull(SkippedEvents.First().SourceResource.Path);
+            Assert.NotNull(SkippedEvents.First().DestinationResource.Path);
+        }
+
+        /// <summary>
+        /// This asserts that the expected events occurred during a single transfer that is expected
+        /// to have a <see cref="StorageTransferStatus.CompletedWithFailedTransfers"/> at the end without any skips.
+        /// </summary>
+        public void AssertSingleFailedCheck()
+        {
+            Assert.IsEmpty(SkippedEvents);
+            Assert.IsEmpty(SingleCompletedEvents);
+            Assert.AreEqual(3, StatusEvents.Count);
+            Assert.AreEqual(1, FailedEvents.Count);
+            Assert.AreEqual(StorageTransferStatus.InProgress, StatusEvents.First().StorageTransferStatus);
+            Assert.AreEqual(StorageTransferStatus.CancellationInProgress, StatusEvents.ElementAt(1).StorageTransferStatus);
+            Assert.AreEqual(StorageTransferStatus.CompletedWithFailedTransfers, StatusEvents.ElementAt(2).StorageTransferStatus);
+            Assert.NotNull(FailedEvents.First().Exception);
+            Assert.NotNull(FailedEvents.First().SourceResource.Path);
+            Assert.NotNull(FailedEvents.First().DestinationResource.Path);
+        }
+
+        /// <summary>
+        /// This asserts that the expected events occurred during a container transfer that is expected
+        /// to have a <see cref="StorageTransferStatus.Completed"/> at the end without any skips
+        /// or failures.
+        /// </summary>
+        /// <param name="blobCount">
+        /// Expected amount of single transfers within the container transfer.
+        /// </param>
+        public void AssertContainerCompletedCheck(int transferCount)
+        {
+            AssertUnexpectedFailureCheck();
+            Assert.IsEmpty(SkippedEvents);
+            Assert.AreEqual(transferCount, SingleCompletedEvents.Count);
```

Comment:
This line is causing the flakiness to occur. Going to disable this for now until we can circle back on a solution on the flakiness of a completed state and getting back all our SingleCompletedEvents.

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/35976